src/content/docs/en/reference/experimental-flags/csp.mdx

```diff
@@ -0,0 +1,379 @@
+    experimental: {
+      csp: true
+    }
+});
+```
+
+When enabled, Astro will add a `<meta>` element inside the `<head>` element of each page.
+
+This element will have the `http-equiv="content-security-policy"` attribute, and the `content` attribute will provide values for the `script-src` and `style-src` [directives](#directives) based on the script and styles used in the page.
+
+```html
+<head>
+    <meta 
+        http-equiv="content-security-policy" 
+        content="
+            script-src 'self' 'sha256-somehash'; 
+            style-src:'self' 'sha256-somehash'"
+   >
+</head>
+```
+
+## Configuration
+
+You can further customize the `<meta>` tag by enabling this feature with a configuration object that includes additional options.
+
+### `algorithm`
+
+<p>
+
+**Type:** `'SHA-256' | 'SHA-512' | 'SHA-384' `<br />
+**Default:** `'SHA-256'`<br />
+<Since v="5.9.0" />
+</p>
+
+The [hash function](https://developer.mozilla.org/en-US/docs/Glossary/Hash_function) to use to generate the hashes of the styles and scripts.
+
+
+```js title="astro.config.mjs"
+import { defineConfig } from 'astro/config';
+
+export default defineConfig({
+    experimental: {
+      csp: {
+        algorithm: 'SHA-512'
+      }
+    }
+});
+```
+
+### `directives`
+
+<p>
+
+**Type:** `[]` <br />
+**Default:** `[]`<br />
+<Since v="5.9.0" />
+</p>
+
+A list of [CSP directives](https://content-security-policy.com/#directive) that defines valid sources for specific content types.
+
+While Astro needs to control the `script-src` and `style-src` directives, it is possible to control other CSP directives using the `csp.directives` field. These directives are added to all pages. It accepts a list of type-safe directives:
+
+```js title="astro.config.mjs"
+import { defineConfig } from 'astro/config';
+
+export default defineConfig({
+    experimental: {
+      csp: {
+        directives: [
+          "default-src 'self'",
+          "image-src 'self' 'https://images.cdn.example.com'"
+        ]
+      }
+    }
+});
+```
+
+After the build, the `<meta>` element will add your directives into the `content` value alongside Astro's default directives:
+
+```html
+<meta
+    http-equiv="content-security-policy"
+    content="
+        default-src 'self'; 
+        image-src 'self' 'https://images.cdn.example.com';
+        script-src 'self' 'sha256-somehash'; style-src:'self' 'sha256-somehash'
+    "
+>
+```
+
+### `stylesDirective` and `scriptsDirective`
+
+<p>
+
+**Type:** `CspDirective` <br />
+**Default:** `{}`<br />
+<Since v="5.9.0" />
+</p>
+
+Configuration objects that allow you to override the default sources for the `style-src` and `script-src` directives with the [`resources`](#resources) property, or to provide additional [hashes](#hashes) to be rendered.
+
+These properties are added to all pages and **completeley override Astro's defaults**, not add to them. Therefore, you must explicitly specify any default values that you want to be included.
```

Comment:
This phrase isn't correct. The override applies only to `resources`, not `hashes`. We should remove it

Reply:
Which phrase exactly? Should it be removed, or amended? Because it does feel important to mention that if you override the default style or script configs, you have nothing and have to include back the default